stablehlo/tests/legalize_stablehlo_to_vhlo.mlir

```diff
@@ -382,11 +382,11 @@ func.func @op_all_reduce(%arg0: tensor<f32>) -> tensor<f32> {
 func.func @op_all_to_all(%arg0: tensor<4x16xf32>) -> tensor<16x4xf32> {
   //               CHECK: "vhlo.all_to_all_v2"(%arg0) {
   //          CHECK-SAME:   channel_handle = #vhlo.channel_handle<handle = 1, type = 0>
-  //          CHECK-SAME:   concat_dimension = 0 : i64,
-  // CHECK-SAME{LITERAL}:   replica_groups = dense<[[0, 1, 2, 3]]> : tensor<1x4xi64>,
-  //          CHECK-SAME:   split_count = 4 : i64,
-  //          CHECK-SAME:   split_dimension = 1 : i64
-  //          CHECK-SAME: } : (tensor<4x16xf32>) -> tensor<16x4xf32>
+  //          CHECK-SAME:   concat_dimension = #vhlo.integer<0 : i64>,
+  // CHECK-SAME{LITERAL}:   replica_groups = #vhlo.dense<dense<[[0, 1, 2, 3]]> : tensor<1x4xi64>>,
```

Comment:
To avoid having to copy the entire implementation for dense printing/parsing, I create a temporary dense array and print/parse:

```
void DenseIntOrFPElementsV1Attr::print(mlir::AsmPrinter& p) const {
  VhloToStablehloTypeConverter conv;
  p << '<'
    << DenseIntOrFPElementsAttr::getFromRawBuffer(conv.convertType(getType()),
                                                  getRawData())
    << '>';
}

Attribute DenseIntOrFPElementsV1Attr::parse(AsmParser& parser, mlir::Type) {
  StablehloToVhloTypeConverter conv;
  DenseIntOrFPElementsAttr attr;
  if (failed(parser.parseLess()) || failed(parser.parseAttribute(attr)) ||
      failed(parser.parseGreater())) {
    return DenseIntOrFPElementsV1Attr();
  }
  return DenseIntOrFPElementsV1Attr::get(
      parser.getContext(), conv.convertType(attr.getType()), attr.getRawData());
}
```

As for "why not wrap", as described above, this would push conversions into the VhloBytecodeInterface code if we want to use the standard `writeType/Attribute` functions.